ikasaneip/harvesting/src/main/java/org/ikasan/harvesting/HarvestingJobImpl.java

```diff
@@ -0,0 +1,230 @@
+        {
+            logger.error("Unable to initialise house keeping job: " + this.getJobName()
+                + ". This may be due to the database not yet having been created.", e);
+        }
+
+        this.initialised = true;
+    }
+
+    @Override
+    public void execute(JobExecutionContext jobExecutionContext) throws JobExecutionException
+    {
+        logger.debug("Harvesting job executing: " + this.getJobName()
+            + " [batch delete size: " + this.harvestSize + "]");
+
+        try
+        {
+            if (harvestService.harvestableRecordsExist())
+            {
+                List<HarvestEvent> events = this.harvestService.harvest(this.harvestSize);
+
+                if(events.size() > 0 && dashboardRestService.publish(events))
+                {
+                    harvestService.updateAsHarvested(events);
+                }
+            }
+
+        }
+        catch(Exception e)
+        {
+            this.executionErrorMessage = e.getMessage();
+            this.lastExecutionSuccessful = false;
+            throw new JobExecutionException("Could not execute housekeeping job: " + this.jobName, e);
+        }
+
+        this.lastExecutionSuccessful = true;
+        logger.debug("Finished harvesting job executing: " + this.getJobName());
+    }
+
+    @Override
+    public void save()
+    {
+    }
+
+
+    public void setCronExpression(String cronExpression)
+    {
+        this.cronExpression = cronExpression;
+    }
+
+    public String getCronExpression()
+    {
+        cronExpression = this.environment.getProperty(this.getJobName() + CRON_EXPRESSION);
+
+        if(cronExpression == null || cronExpression.isEmpty())
+        {
+            cronExpression = DEFAULT_CRON_EXPRESSION;
+        }
+
+        return cronExpression;
+    }
+
+
+    public Environment getEnvironment()
+    {
+        return environment;
+    }
+    @Override
+    public String getJobName()
+    {
+        return jobName;
+    }
+    @Override
+    public Integer getHarvestSize()
+    {
+        return harvestSize;
+    }
+    @Override
+    public void setHarvestSize(Integer harvestSize)
+    {
+        this.harvestSize = harvestSize;
+    }
+    @Override
+    public Boolean isEnabled()
+    {
+        return enabled;
+    }
+    @Override
+    public void setEnabled(Boolean enabled)
+    {
+        this.enabled = enabled;
+    }
+    @Override
+    public Boolean getLastExecutionSuccessful()
+    {
+        return lastExecutionSuccessful;
+    }
+    @Override
+    public String getExecutionErrorMessage()
+    {
+        return executionErrorMessage;
+    }
+    @Override
+    public Boolean isInitialised()
+    {
+        return initialised;
+    }
+    @Override
+    public void setInitialised(Boolean initialised)
+    {
+        this.initialised = initialised;
+    }
+
+    @Override public Integer getThreadCount()
+    {
+        return null;
+    }
+
+    @Override public void setThreadCount(Integer threadCount)
```

Comment:
why empty implementation?

Reply:
Above implementation is specific to the module and does not use multithreading. Dashboard Job implementation utilising solr is multithreaded.  One option would be to remove the method from the interface but that Dashboard code would need to be made aware of the implementation aka not coding agains interface.

Reply:
maybe just stick a comment in the method to the effect of "not required"